doc/source/whatsnew/v0.18.1.txt

```diff
@@ -76,8 +76,7 @@ API changes
 
 
 
-
-
+- In ``read_csv``, the Python parser now raises an ``EmptyDataError`` in response to empty columns or header (:issue:`12506`)
```

Comment:
are you sure? show me an explicit example. This is the point, we are not making ANY API changes unless they are very explicit. I suspect some API is not actually tested. In that case you have to make doubly sure by constructing tests whenever you change things (and then of course add the tests).

My entire point of this issue is to really minimize the user changes if any. At the same time we have a fundamental disconnect between c-engine and python-engine. We are changing from raiing ``StopIteration`` on *some* errors to a ``ValueError``. So this is  small API change that will have to be shown with a complete example. But I don't think we really have coverage of which errors are raised when, so need to sort them out and see what potential changes are. Then we can just whether to accept them now (or defer or whatever).